Update 3.4.0 release notes (#649)
pynamodb/PynamoDB: docs/release_notes.rst

Release Notes
=============

v3.4.0
------

:date: 2019-06-13

This is a backwards compatible, minor release.

Changes in this release:

>  * Adds a TTLAttribute that dynamo uses to delete records

Reviewer: ```suggestion
 * Adds a TTLAttribute that specifies when items expire
```

>  * Enables TTL deletion on a dynamo table if the corresponding model has a TTLAttribute

Reviewer: ```suggestion
 * Enables time-to-live on a DynamoDB table if the corresponding model has a TTLAttribute
```

>  * default_for_new sets default values only for unsaved objects

Reviewer: ```suggestion
 * Adds a default_for_new parameter to Attribute which is a default that applies to new items only
```

Author: Is this too much detail?

Reviewer: I agree, let's ~remove~ the part in parenthesis
